src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/PooledHashSetExtensions.cs

```diff
@@ -68,42 +72,95 @@ internal static class PooledHashSetExtensions
                 isInterface: isInterface,
                 taintedProperties: taintedProperties?.ToImmutableHashSet(StringComparer.Ordinal)
                     ?? ImmutableHashSet<string>.Empty,
-                taintedMethodsNeedPointsToAnalysis:
+                 taintedMethodsNeedsPointsToAnalysis:
                     taintedMethods
-                    ?.Select(o => new KeyValuePair<string, IsInvocationTaintedWithPointsToAnalysis>(o, (IEnumerable<PointsToAbstractValue> argumentValueContents) => { return true; }))
-                    ?.ToImmutableDictionary(StringComparer.Ordinal)
-                    ?? ImmutableDictionary<string, IsInvocationTaintedWithPointsToAnalysis>.Empty,
+                        ?.Select(o => new KeyValuePair<string, PointsToChecksAndTargets>(
+                            o,
+                            PointsToChecksAndTargets.Empty.Add(
+                                (IEnumerable<IArgumentOperation> arguments, IEnumerable<PointsToAbstractValue> argumentPointsTo) => { return true; },
+                                new[] { "RETURN" }.ToImmutableHashSet())))
+                        ?.ToImmutableDictionary(StringComparer.Ordinal)
+                    ?? ImmutableDictionary<string, PointsToChecksAndTargets>.Empty,
                 taintedMethodsNeedsValueContentAnalysis:
-                    ImmutableDictionary<string, IsInvocationTaintedWithValueContentAnalysis>.Empty,
+                    ImmutableDictionary<string, ValueContentChecksAndTargets>.Empty,
                 taintConstantArray: false);
             builder.Add(metadata);
         }
 
+        // Just to make hardcoding SourceInfos more convenient.
+        public static void AddSourceInfoSpecifyingTaintedTargets(
```

Comment:
Add /// XML documentation.  Since the taintedMethodsNeedsPointsToAnalysis and taintedMethodsNeedsValueContentAnalysis arguments aren't straightforward, need to explain what the "taintedTargets" are.